### DISCUSSION TITLE
Strongswan : does strongswan provides log for ESN overflow

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2457] [Q&A]
What is the size of ESN supported by strongswan ? Is it 32 or 64 bit? Is there any log event if ESN overflow occurs?

**Comment by tobiasbrunner** [comment]
What IPsec implementation are you using? And ESN are always 64 bit.

**Reply by hnsantu554** [reply]
I am using ESN for ESP protocol with esp_proposals = aes128gcm128-esn

Is there any log event from strongswan if the ESN overflow occurs?

**Reply by tobiasbrunner** [reply]
If IPsec is handled by the Linux kernel, it will generate an `SA-replay-overflow` audit message (no matter if ESN is used or not) if the sequence number overflows when sending a packet. `XfrmOutStateSeqError` is also increased in `/proc/net/xfrm_stat`. strongSwan won't know anything about that.
